src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -116,15 +140,23 @@ public void storeMessage(Message message) {
       Message message = new Message(id, sender, text, timestamp, receiver);
       messages.add(message);
     }
-
     return messages;
-  }
-
-  /** Returns the total number of messages for all users. */
-  public int getTotalMessageCount() {
-    Query query = new Query("Message");
-    PreparedQuery results = datastore.prepare(query);
-    return results.countEntities(FetchOptions.Builder.withDefaults());
-  }
+ }
+
+ private boolean hasUserRestriction(String loggedInUser, String otherUser) {
+   if (!Strings.isNullOrEmpty(loggedInUser) && !Strings.isNullOrEmpty(otherUser)) {
+     return false;
+   }
+   return true;
+ }
+ private Query createMessageQueryWithUserFilter(String loggedInUser, String otherUser, Filter directMessages) {
```

Comment:
This function does not use the `loggedInUser` or `otherUser` parameters. Why are they passed-in?

My recommendation would be:

```
   private Query createMessageQueryWithUserFilter(String loggedInUser, String otherUser) {
      return Query("Message")
         .setFilter(createUserFilter(loggedInUser, otherUser))
         .addSort("timestamp", SortDirection.ASCENDING);
   }
```

Reply:
Created a new function ```createUserFilter(loggedInUser, otherUser))```
Thank you for your recommendation